index.js

```diff
@@ -154,17 +154,29 @@ var MongodbDriver = Base.extend({
    * Adds an index to a collection
    *
    * @param collectionName  - The collection to add the index to
-   * @param indexName       - The name of the index to add
-   * @param columns         - The columns to add an index on
-   * @param	unique          - A boolean whether this creates a unique index
+   * @param indexOptions    - An object of options to be used as Index options
+   * @param callback
    */
-  addIndex: function(collectionName, indexName, columns, unique, callback) {
-
-    var options = {
-      indexName: indexName,
-      columns: columns,
-      unique: unique
-    };
+  addIndex: function(collectionName, indexOptions, callback) {
+    var options = {};
+    if (indexOptions.constructor.name == 'Object') {
+      options = {
+        name: indexOptions.name,
+        columns: indexOptions.columns,
+        unique: indexOptions.unique
+      };
+      if (indexOptions.hasOwnProperty('sparse')) {
```

Comment:
Ok, lets be a bit more specific: What is the default value of sparse? If it is `false`, then it can just be omitted. If the default is true, well then please check for `indexOptions.sparse === false` instead.